src/sinks/util/service.rs

```diff
@@ -125,11 +125,35 @@ impl<'de> Deserialize<'de> for InFlightLimit {
     }
 }
 
+pub trait InFlightLimitOption {
+    fn unwrap_with(&self, default: &Self) -> Option<usize>;
+}
```

Comment:
```suggestion
pub trait ParseInFlightLimit {
    fn parse_in_flight_limit(&self, default: &Self) -> Option<usize>;
}
```

Reply:
Just now I realized that instead of custom trait I can use [std::convert::Into](https://doc.rust-lang.org/std/convert/trait.Into.html).